lib/srv/discovery/discovery.go

```diff
@@ -416,7 +418,10 @@ func (s *Server) handleEC2Discovery() {
 				instances.AccountID, genEC2InstancesLogStr(ec2Instances.Instances))
 
 			if err := s.handleEC2Instances(ec2Instances); err != nil {
-				if trace.IsNotFound(err) {
+				var aErr awserr.Error
+				if errors.As(err, &aErr) && aErr.Code() == ssm.ErrCodeInvalidInstanceId {
+					s.Log.WithError(err).Error("Invalid instance ID found. This can happen if the instance doesnt have a running SSM agent that is registered with the SSM endpoint (may require reinstalling the SSM Agent, or giving the instance IAM permissions to receive SSM commands), or the discovery instance doesnt have permissions to access the node.")
```

Comment:
Suggestion to reduce unnecessary words:
```suggestion
					s.Log.WithError(err).Error("Invalid instance ID found. This can happen if the instance doesnt have a running SSM agent registered with the SSM endpoint (may require reinstalling the SSM Agent, or giving the instance IAM permissions to receive SSM commands), or the discovery instance doesnt have permissions to access the node.")
```

And can you use punctuation for the contraction, or will a `'` escape the string?

Reply:
alternatively, remove the contraction: s/doesnt/does not/g